packages/matrix-client-server/src/account/3pid/delete.ts

```diff
@@ -16,15 +23,126 @@ interface RequestBody {
 interface RegisterResponseBody {
   token: string
 }
+
+interface OpenIDResponseBody {
+  access_token: string
+  expires_in: number
+  matrix_server_name: string
+  token_type: string
+}
+
 const schema = {
   address: true,
   id_server: false,
   medium: true
 }
 
+const deleteAndSend = async (
+  res: e.Response | ServerResponse,
+  body: RequestBody,
+  clientServer: MatrixClientServer,
+  idServer: string,
+  data: TokenContent,
+  token: string | null
+): Promise<void> => {
+  const openIDResponse = await fetch(
+    `https://${clientServer.conf.server_name}/_matrix/client/v3/user/${data.sub}/openid/request_token`,
+    {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${token as string}`
+      },
+      body: JSON.stringify({
+        user_id: data.sub
+      })
+    }
+  )
+  // istanbul ignore if
+  if (!openIDResponse.ok) {
+    // istanbul ignore next
+    send(
+      res,
+      500,
+      errMsg('unknown', 'Error while requesting OpenID token'),
+      clientServer.logger
+    )
+    // istanbul ignore next
+    return
+  }
+  const openIDResponseBody: OpenIDResponseBody =
+    (await openIDResponse.json()) as OpenIDResponseBody
+
+  const matrixResolve = new MatrixResolve({
+    cache: 'toad-cache'
+  })
+  const baseUrl: string | string[] = await matrixResolve.resolve(idServer)
+  const registerResponse = await fetch(
+    `https://${baseUrl as string}/_matrix/identity/v2/account/register`,
+    {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(openIDResponseBody)
+    }
+  )
+  const validToken = (
+    registerResponse.json() as unknown as RegisterResponseBody
```

Comment:
Should be `(await registerResponse.json()) as unknown as RegisterResponseBody` if I'm right in previous comment

Reply:
Yes you're right I've fixed it thank you !